core/src/main/java/bio/terra/pearl/core/service/dataimport/ImportService.java

```diff
@@ -2,16 +2,63 @@
 
 import bio.terra.pearl.core.dao.dataimport.ImportDao;
 import bio.terra.pearl.core.model.dataimport.Import;
+import bio.terra.pearl.core.model.dataimport.ImportItemStatus;
+import bio.terra.pearl.core.model.dataimport.ImportStatus;
 import bio.terra.pearl.core.service.CrudService;
+import bio.terra.pearl.core.service.admin.AdminUserService;
+import bio.terra.pearl.core.service.exception.NotFoundException;
+import bio.terra.pearl.core.service.participant.EnrolleeService;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
 @Service
 @Slf4j
 public class ImportService extends CrudService<Import, ImportDao> {
 
+    @Autowired
+    ImportItemService importItemService;
+
+    @Autowired
+    AdminUserService adminUserService;
+
+    @Autowired
+    EnrolleeService enrolleeService;
+
     public ImportService(ImportDao dao) {
         super(dao);
     }
 
+    public List<Import> findAllByStudyEnv(UUID studyEnvId) {
+        List<Import> imports = dao.findAllByStudyEnv(studyEnvId);
+        //load ImportItems
+        imports.forEach(anImport -> importItemService.attachImportItems(anImport));
+        return imports;
+    }
+
+    public Import updateStatus(UUID id, ImportStatus status) {
+        Import dataImport = dao.find(id).orElseThrow(() -> new NotFoundException("Import not found "));
+        dataImport.setStatus(status);
+        //update all importItems
+        importItemService.updateStatusByImportId(id, ImportItemStatus.DELETED);
+        return dao.update(dataImport);
+    }
+
+    public void deleteEnrolleesByImportId(UUID id) {
```

Comment:
Wondering if it makes sense to add @Transactional in *ExtService in some cases , since we might be calling multiple methods in the ImportService
ex:- EnrolleeImportExtService.delete

Reply:
yeah, if the Ext service wants the operation to succeed/fail as a whole, we should use Transactional in the Ext service